Libraries/Opc.Ua.Client/Session/Session.cs

```diff
@@ -3652,6 +3745,279 @@ public bool RemoveTransferredSubscription(Subscription subscription)
+                                includeSubtypes,
+                                nodeClassMask,
+                                out var resultForBatch,
+                                out var errorsForBatch
+                            );
+
+                        for (int ii = 0; ii < nodesToBrowseBatch.Count; ii++)
+                        {
+                            if(errorsForBatch[ii].StatusCode == StatusCodes.BadNoContinuationPoints ||
+                                errorsForBatch[ii].StatusCode == StatusCodes.BadContinuationPointInvalid)
+                            {
+                                nodesToBrowseForNextPass.Add( nodesToBrowseForPass[batchOffset + ii] );                                
+                                referenceDescriptionsForNextPass.Add( resultForPass[batchOffset + ii] );                                
+                                errorsForNextPass.Add( errorsForPass[batchOffset + ii] );
+                            }
+
+                            resultForPass[batchOffset + ii].Clear();
+                            resultForPass[batchOffset + ii].AddRange( resultForBatch[ii] );
+                            errorsForPass[batchOffset + ii] = errorsForBatch[ii];
+                            
+                        }
+                        int badNoCp = errorsForBatch.Count(x => x.StatusCode == StatusCodes.BadNoContinuationPoints);
+                        int badCpI = errorsForBatch.Count(x => x.StatusCode== StatusCodes.BadContinuationPointInvalid);
+                        int bad = errorsForBatch.Count(x => StatusCode.IsBad(x.StatusCode));
+                        badNoCPErrorsPerPass += badNoCp;
+                        badCPInvalidErrorsPerPass += badCpI;
+                        otherErrorsPerPass += bad - badNoCp - badCpI;
+
+                        batchCount++;
+                    }
+
+                    resultForPass = new List<ReferenceDescriptionCollection>();
+                    resultForPass.AddRange( referenceDescriptionsForNextPass );
+                    referenceDescriptionsForNextPass.Clear();
+
+                    errorsForPass = new List<ServiceResult>();
+                    errorsForPass.AddRange( errorsForNextPass );
+                    errorsForNextPass.Clear();
+
+                    nodesToBrowseForPass = new List<NodeId>();
+                    nodesToBrowseForPass.AddRange( nodesToBrowseForNextPass );
+                    nodesToBrowseForNextPass.Clear();
+
+                    String aggregatedErrorMessage = "ManagedBrowse: in pass {0}, {1} {2} occured with a status code {3}";
+
+                    if (badCPInvalidErrorsPerPass > 0)
+                    {                         
+                        Utils.LogInfo(aggregatedErrorMessage, passCount, badCPInvalidErrorsPerPass,
+                            badCPInvalidErrorsPerPass == 1 ? "error" : "errors", "BadContinuationPointInvalid.");
+                    }
```

Comment:
is it possible to use nameof() for the StatusCode string?

Reply:
Yes, that's possible.